lib/grape/api/instance.rb

```diff
@@ -121,7 +121,7 @@ def evaluate_as_instance_with_configuration(block, lazy: false)
             self.configuration = value_for_configuration
             response
           end
-          if base_instance? && lazy
+          if base && base_instance? && lazy
```

Comment:
Should we be doing this check inside `base_instance?`

Reply:
I don't think it makes sense, as `base_instance?` usually means: "This instance is not used anywhere and is just a template for other instances" while, if an API has no `base`, it will most likely be used directly